codegen/type_converter.go

```diff
@@ -346,16 +346,22 @@ func (c *TypeConverter) genConverterForMap(
 		return err
 	}
 
+	valueStruct, isStruct := toFieldType.ValueSpec.(*compile.StructSpec)
+	sourceIdentifier := fromIdentifier
 	_, isStringKey := toFieldType.KeySpec.(*compile.StringSpec)
+
 	if !isStringKey {
-		return errors.Errorf(
```

Comment:
We need a new check here. We do not want to generate code for `map<map<string,string>, string>` 

The key should be a `string` or a typedef for a `string`.

There is a `compile.RootTypeSpec()` method that will take a typedef and give you back the actual type it is.